### PR TITLE
fix: fix LN deprecation warnings

### PR DESCRIPTION
**Pull request by m1sterc001guy**
https://github.com/fedimint/fedimint/pull/4024 added some deprecation warnings. This PR just suppresses them since we don't need to see them during build.

### CODE REVIEW

File: modules/fedimint-ln-client/src/pay.rs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+#![allow(deprecated)]
```
Comment (dpc) [inline modules/fedimint-ln-client/src/pay.rs:1]: A crate level `allow` can let multiple new ones slip it, and also no one will remember to clean it up after it is no longer needed. IMO, we should put it on every line that is using the deprecated symbol.

Comment (m1sterc001guy) [reply modules/fedimint-ln-client/src/pay.rs:1]: So I tried that initially, and I could only get the usage warning to go away, but for some reason when adding `#[allow(deprecated)]` on the declaration, the derive statement on the enum causes problems and the `#[allow(deprecated)]` macro is ignored. 

Maybe that's good enough?

Comment (dpc) [reply modules/fedimint-ln-client/src/pay.rs:1]: Let's try what we for start.

**Review by dpc — APPROVED** [review]

File: modules/fedimint-ln-client/src/pay.rs [hunk 2]
```diff
@@ -26,42 +27,53 @@ use crate::{set_payment_result, LightningClientContext, LightningClientStateMach
...
-    Refundable(LightningPayRefundable),
-    Refund(LightningPayRefund),
-    #[deprecated(
-        since = "0.4.0",
-        note = "Pay State Machine does not need to wait for the refund tx to be accepted"
-    )]
-    Refunded(Vec<OutPoint>),
-    Failure(String),
+#[allow(deprecated)]
+pub mod lightningpay {
```
Comment (dpc) [inline modules/fedimint-ln-client/src/pay.rs:31]: `pub(crate)` or `pub(super)`.

**Review by dpc — CHANGES_REQUESTED** [review]
actually, nits.

File: modules/fedimint-ln-client/src/pay.rs (hunk 2 above)
Comment (dpc) [reply modules/fedimint-ln-client/src/pay.rs:31]: And a comment why we need this module.

Comment (m1sterc001guy) [reply modules/fedimint-ln-client/src/pay.rs]: done

Comment (m1sterc001guy) [reply modules/fedimint-ln-client/src/pay.rs]: ah unfortunately needs to be `pub` since it is used in the testing crate

**Review by dpc — APPROVED** [review]

**Review by dpc — APPROVED** [review]

**Pull request merged**